src/build/emitter.ts

```diff
@@ -828,12 +828,24 @@ export function emitWebIdl(
   ) {
     if (entity.comment) {
       if (entity.comment.startsWith("/*")) {
-        entity.comment.split("\n").forEach(print);
+        const split = entity.comment.split("\n");
+
+        // Merge in the deprecated notice, if required.
+        if (entity.deprecated && !entity.comment.includes("@deprecated")) {
+          split[split.length - 1] = " * @deprecated";
+          split.push(" */");
+        }
+
+        split.forEach(print);
       } else {
-        print(`/** ${entity.comment} */`);
+        if (entity.deprecated && !entity.comment.includes("@deprecated")) {
```

Comment:
formatting: you can merge `else { if (...` to `else if (...` and drop an indent level

Reply:
Done